src/backend/distributed/planner/multi_explain.c

```diff
@@ -1353,8 +1353,8 @@ ExplainOneQuery(Query *query, int cursorOptions,
 		INSTR_TIME_SUBTRACT(planduration, planstart);
 
 		/* run it (if needed) and produce output */
-		ExplainOnePlan(plan, into, es, queryString, params, queryEnv,
-					   &planduration);
+		ExplainOnePlanCompat(plan, into, es, queryString, params, queryEnv,
+					   &planduration, NULL);
```

Comment:
I think passing `NULL` here breaks things. When citus is in the preload libraries (created and dropped) I get the following output:

```
[local] nilsdijk@postgres:9700-20770=# explain (analyze, buffers) select * from pg_class WHERE relname LIKE 't%';
┌───────────────────────────────────────────────────────────────────────────────────────────────────────────────────────────────────┐
│                                                            QUERY PLAN                                                             │
├───────────────────────────────────────────────────────────────────────────────────────────────────────────────────────────────────┤
│ Bitmap Heap Scan on pg_class  (cost=4.34..16.65 rows=9 width=265) (actual time=0.066..0.069 rows=6 loops=1)                       │
│   Filter: (relname ~~ 't%'::text)                                                                                                 │
│   Heap Blocks: exact=1                                                                                                            │
│   Buffers: shared hit=5 read=1                                                                                                    │
│   ->  Bitmap Index Scan on pg_class_relname_nsp_index  (cost=0.00..4.34 rows=7 width=0) (actual time=0.033..0.033 rows=6 loops=1) │
│         Index Cond: ((relname >= 't'::text) AND (relname < 'u'::text))                                                            │
│         Buffers: shared hit=5                                                                                                     │
│ Planning Time: 58.563 ms                                                                                                          │
│ Execution Time: 0.808 ms                                                                                                          │
└───────────────────────────────────────────────────────────────────────────────────────────────────────────────────────────────────┘
(9 rows)

Time: 77.555 ms
```

When I then remove citus from the preload libraries and restart the output is:
```
[local] nilsdijk@postgres:9700-18472=# explain (analyze, buffers) select * from pg_class WHERE relname LIKE 't%';
┌───────────────────────────────────────────────────────────────────────────────────────────────────────────────────────────────────┐
│                                                            QUERY PLAN                                                             │
├───────────────────────────────────────────────────────────────────────────────────────────────────────────────────────────────────┤
│ Bitmap Heap Scan on pg_class  (cost=4.34..16.65 rows=9 width=265) (actual time=0.059..0.061 rows=6 loops=1)                       │
│   Filter: (relname ~~ 't%'::text)                                                                                                 │
│   Heap Blocks: exact=1                                                                                                            │
│   Buffers: shared hit=5 read=1                                                                                                    │
│   ->  Bitmap Index Scan on pg_class_relname_nsp_index  (cost=0.00..4.34 rows=7 width=0) (actual time=0.030..0.030 rows=6 loops=1) │
│         Index Cond: ((relname >= 't'::text) AND (relname < 'u'::text))                                                            │
│         Buffers: shared hit=5                                                                                                     │
│ Planning Time: 36.069 ms                                                                                                          │
│   Buffers: shared hit=159 read=30                                                                                                 │
│ Execution Time: 0.756 ms                                                                                                          │
└───────────────────────────────────────────────────────────────────────────────────────────────────────────────────────────────────┘
(10 rows)

Time: 63.027 ms
```

Reply:
https://github.com/citusdata/citus/pull/3900/commits/8357ffeb41f643eca7ff5f22979130e36e30c918

Reply:
There is some duplicate with https://github.com/citusdata/citus/pull/3900/commits/8357ffeb41f643eca7ff5f22979130e36e30c918, I might add a refactor commit as well.

Reply:
This refactoring is more related to the original code, so I haven't done the refactoring but followed the original code for now.